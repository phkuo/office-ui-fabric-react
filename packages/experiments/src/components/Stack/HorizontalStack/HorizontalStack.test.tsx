import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import { HorizontalStack } from '../index';

describe('HorizontalStack', () => {
  it('can handle having no children', () => {
    const createEmptyStack = () => {
      mount(<HorizontalStack />);
    };

    expect(createEmptyStack).not.toThrow();
  });

  it('renders default HorizontalStack correctly', () => {
    const component = renderer.create(
      <HorizontalStack>
        <div>Item 1</div>
        <div>Item 2</div>
      </HorizontalStack>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders gap correctly', () => {
    const component = renderer.create(
      <HorizontalStack gap={5}>
        <div>Item 1</div>
        <div>Item 2</div>
      </HorizontalStack>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders growing items correctly', () => {
    const component = renderer.create(
      <HorizontalStack>
        <HorizontalStack.Item>Item 1</HorizontalStack.Item>
        <HorizontalStack.Item>Item 2</HorizontalStack.Item>
      </HorizontalStack>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders shrinking items correctly', () => {
    const component = renderer.create(
      <HorizontalStack shrinkItems>
        <HorizontalStack.Item>Item 1</HorizontalStack.Item>
        <HorizontalStack.Item preventShrink>Item 2</HorizontalStack.Item>
      </HorizontalStack>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders item alignments correctly', () => {
    const component = renderer.create(
      <HorizontalStack>
        <HorizontalStack.Item align="auto">Auto</HorizontalStack.Item>
        <HorizontalStack.Item align="stretch">Stretch</HorizontalStack.Item>
        <HorizontalStack.Item align="baseline">Baseline</HorizontalStack.Item>
        <HorizontalStack.Item align="start">Start</HorizontalStack.Item>
        <HorizontalStack.Item align="center">Center</HorizontalStack.Item>
        <HorizontalStack.Item align="end">End</HorizontalStack.Item>
      </HorizontalStack>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders wrapped HorizontalStack correctly', () => {
    const component = renderer.create(
      <HorizontalStack wrap>
        <HorizontalStack.Item>1</HorizontalStack.Item>
        <HorizontalStack.Item>2</HorizontalStack.Item>
        <HorizontalStack.Item>3</HorizontalStack.Item>
      </HorizontalStack>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
