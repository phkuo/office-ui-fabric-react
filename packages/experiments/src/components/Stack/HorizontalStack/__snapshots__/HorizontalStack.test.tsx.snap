// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalStack renders default HorizontalStack correctly 1`] = `
<div
  className=
      ms-HorizontalStack
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        box-sizing: border-box;
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        height: auto;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
        white-space: nowrap;
      }
      & > *:not(:empty) ~ * {
        margin-left: 0px;
      }
      & > *:empty {
        display: none;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 0;
      }
>
  <div>
    Item 1
  </div>
  <div>
    Item 2
  </div>
</div>
`;

exports[`HorizontalStack renders gap correctly 1`] = `
<div
  className=
      ms-HorizontalStack
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        box-sizing: border-box;
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        height: auto;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
        white-space: nowrap;
      }
      & > *:not(:empty) ~ * {
        margin-left: 5px;
      }
      & > *:empty {
        display: none;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 0;
      }
>
  <div>
    Item 1
  </div>
  <div>
    Item 2
  </div>
</div>
`;

exports[`HorizontalStack renders growing items correctly 1`] = `
<div
  className=
      ms-HorizontalStack
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        box-sizing: border-box;
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        height: auto;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
        white-space: nowrap;
      }
      & > *:not(:empty) ~ * {
        margin-left: 0px;
      }
      & > *:empty {
        display: none;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 0;
      }
>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Item 1
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Item 2
  </span>
</div>
`;

exports[`HorizontalStack renders item alignments correctly 1`] = `
<div
  className=
      ms-HorizontalStack
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        box-sizing: border-box;
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        height: auto;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
        white-space: nowrap;
      }
      & > *:not(:empty) ~ * {
        margin-left: 0px;
      }
      & > *:empty {
        display: none;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 0;
      }
>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: auto;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Auto
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: stretch;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Stretch
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: baseline;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Baseline
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: flex-start;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Start
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: center;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Center
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-self: flex-end;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    End
  </span>
</div>
`;

exports[`HorizontalStack renders shrinking items correctly 1`] = `
<div
  className=
      ms-HorizontalStack
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        box-sizing: border-box;
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        height: auto;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
        white-space: nowrap;
      }
      & > *:not(:empty) ~ * {
        margin-left: 0px;
      }
      & > *:empty {
        display: none;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 1;
      }
>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          flex-shrink: 1;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Item 1
  </span>
  <span
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          flex-shrink: 0;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    Item 2
  </span>
</div>
`;

exports[`HorizontalStack renders wrapped HorizontalStack correctly 1`] = `
<div
  className=
      ms-HorizontalStack
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        display: inline-block;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        height: auto;
        overflow: visible;
        width: auto;
      }
>
  <div
    className=
        ms-HorizontalStack-inner
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          box-sizing: border-box;
          display: flex;
          flex-direction: row;
          flex-wrap: wrap;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          margin-bottom: 0px;
          margin-left: 0px;
          margin-right: 0px;
          margin-top: 0px;
          max-width: 100vw;
          overflow: visible;
          width: auto;
        }
        & > * {
          margin-bottom: 0px;
          margin-left: 0px;
          margin-right: 0px;
          margin-top: 0px;
          max-width: 100%;
          text-overflow: ellipsis;
          white-space: nowrap;
        }
        & > *:empty {
          display: none;
        }
        & > *:not(.ms-StackItem) {
          flex-shrink: 0;
        }
  >
    <span
      className=
          ms-StackItem
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            flex-shrink: 0;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 14px;
            font-weight: 400;
            height: auto;
            width: auto;
          }
    >
      1
    </span>
    <span
      className=
          ms-StackItem
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            flex-shrink: 0;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 14px;
            font-weight: 400;
            height: auto;
            width: auto;
          }
    >
      2
    </span>
    <span
      className=
          ms-StackItem
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            flex-shrink: 0;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 14px;
            font-weight: 400;
            height: auto;
            width: auto;
          }
    >
      3
    </span>
  </div>
</div>
`;
